src/utils/validation.js

```diff
@@ -0,0 +1,123 @@
+import { jsonDeepCopy } from '.';
+import { PUBLISHED } from '../data/constants';
+
+
+const requiredMessage = 'This field is required';
+
+// Basic validation that ensures some value was entered
+const basicValidate = value => (value ? undefined : requiredMessage);
+
+/**
+ * Iterates through errors on a form and returns the first field name with an error.
+ *
+ * @param {?object} errors - Object with keys giving the field names with errors in a form like:
+ *   {
+ *     short_description: "Required",
+ *     outcome: "Required",
+ *     course_runs: [
+ *       null,
+ *       {
+ *         transcript_languages: {
+ *           _error: "Required",
+ *         },
+ *         staff: "Required"
+ *       }
+ *     ]
+ *   };
+ *   The param will be null if there are no errors.
+ *
+ * @returns {string} - The first field name with a validation error
+ */
+const getFieldName = (errors) => {
+  if (errors) {
```

Comment:
optional nit: I usually write this kind of function inverse style with an early exit. So `if (!errors) return null;`  That way you don't need an extra layer of indenting for the real logic.

Reply:
smart smart